Update dependency check to handle the new version format

jandubois opened the pull request:
The version is now `desktop-vX.Y.Z.rdN`, so can also use the shared `rcompareVersions` function to compare versions with the `.rdN` suffix.

mook-as commented on scripts/dependencies/lima.ts:
```diff
@@ -28,7 +28,7 @@ import { simpleSpawn } from 'scripts/simple_process';
  * @param version2 The second version to compare.
  * @returns Whether version1 is higher (-1), equal to (0), or lower than (1) version2.
  */
-function rcompareVersions(version1: string, version2: string): -1 | 0 | 1 {
```
Might be worthwhile to move this function into somewhere more general; putting it under `lima` is confusing now.

jandubois replied:
Into `scripts/lib/dependencies.ts`?

mook-as replied:
Yeah, that sounds reasonable.

mook-as commented on scripts/dependencies/tools.ts:
```diff
@@ -4,12 +4,23 @@ import path from 'path';
+import { rcompareVersions } from './lima';
 import {
-  download, downloadZip, downloadTarGZ, getResource, DownloadOptions, ArchiveDownloadOptions,
+  ArchiveDownloadOptions,
```
We should see if we can get a linter to do your preferred style of imports…

I think we are already using [https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md#named](`eslint-plugin-import`) so we can just turn the relevant option on (in a different PR)?

mook-as commented on scripts/dependencies/tools.ts:
```diff
@@ -386,34 +396,17 @@ export class RancherDashboard implements Dependency, GitHubDependency {
-    // plugins development. For more info please see
-    // https://github.com/rancher-sandbox/rancher-desktop/issues/3757
-    return versions.filter(version => !version.includes('.plugins.'));
+    return tagNames.map((tagName: string) => tagName.replace(/^desktop-v/, ''));
```
Do we still need to filter out `.plugins.`? Why not?

jandubois replied:
IIUC the old idea was to turn the dashboard into a plugin and have Rancher Desktop be a shell that this would plug into.

It is my understanding that that approach has been abandoned, and the code in the new repo is a "standalone extension". So the new repo will only have releases of this extension. @rak-phillip can correct me if I'm wrong.

rak-phillip replied:
> It is my understanding that that approach has been abandoned, and the code in the new repo is a "standalone extension". So the new repo will only have releases of this extension. @rak-phillip can correct me if I'm wrong.

You are correct.

jandubois committed:
Update dependency check to handle the new version format

The version is now `desktop-vX.Y.Z.rdN`, so can also use the shared
rcompareVersions function to compare versions with the `.rdN` suffix.

Signed-off-by: Jan Dubois <jan.dubois@suse.com>
